### ISSUE TITLE
RateLimiter Exception 

### ISSUE THREAD

**Issue by zhavir** [recurly/recurly-client-python#388]
When I reach the rate-limit I get a no-explicit error message.

The error that I get is the following:
`
self.rate_limit = int(self.__headers.get("X-RateLimit-Limit"))
TypeError: int() argument must be a string, a bytes-like object or a number, not 'NoneType'
`

**Your Environment**

- 3.4.2
- python3.8

**Comment by bhelx** [comment]
Thanks @zhavir

I'll see if I can reproduce. I'm noticing this code that parses the response metadata could be more defensive. We'll get a fix for you and add some more tests around that part of the code.

**Comment by bhelx** [comment]
Was able to repro in test:

```
======================================================================
ERROR: test_cast_from_response_without_rate_limit (tests.test_resource.TestResource)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/beckel/code/recurly/v3/clients/python/tests/test_resource.py", line 83, in test_cast_from_response_without_rate_limit
    empty = cast({}, "Empty", Response(resp, request))
  File "/Users/beckel/code/recurly/v3/clients/python/recurly/response.py", line 19, in __init__
    self.rate_limit = int(self.__headers.get("X-RateLimit-Limit"))
TypeError: int() argument must be a string, a bytes-like object or a number, not 'NoneType'
```

**Comment by bhelx** [comment]
@zhavir Do you mind trying out the change on master now? This fix will get picked up in the next release unless you have an issue with it. Thanks again for supporting!

**Comment by zhavir** [comment]
@bhelx, thanks for working on it ASAP, I'd like to get this fix on a new version, we're minding on update our production library version. Thanks a lot!

**Comment by zhavir** [comment]
@bhelx any news on this? do I need to open a ticket on the Recurly side in order to get the latest version?

**Comment by bhelx** [comment]
@zhavir we don't cut releases too frequently, maybe once a week. I'll cut one for you now!

**Comment by bhelx** [comment]
@zhavir https://github.com/recurly/recurly-client-python/releases/tag/3.5.1
@douglasmiller has released this in 3.5.1. Should be hitting [pypi](https://pypi.org/project/recurly/3.5.1/) now.
